openc3/spec/interfaces/http_client_interface_spec.rb

```diff
@@ -42,7 +58,225 @@ module OpenC3
         expect(i.connection_string).to eql "http://127.0.0.1:8080"
       end
     end
+=end
+
+    before(:each) do
+      @interface = HttpClientInterface.new('example.com', 8080, 'https', 10, 15, 5, true)
+    end
+
+    describe "#initialize" do
+      it "initializes with default parameters" do
+        interface = HttpClientInterface.new('example.com')
+        expect(interface.instance_variable_get(:@hostname)).to eq('example.com')
+        expect(interface.instance_variable_get(:@port)).to eq(80)
+        expect(interface.instance_variable_get(:@protocol)).to eq('http')
+      end
+
+      it "initializes with custom parameters" do
+        expect(@interface.instance_variable_get(:@hostname)).to eq('example.com')
+        expect(@interface.instance_variable_get(:@port)).to eq(8080)
+        expect(@interface.instance_variable_get(:@protocol)).to eq('https')
+        expect(@interface.instance_variable_get(:@write_timeout)).to eq(10.0)
+        expect(@interface.instance_variable_get(:@read_timeout)).to eq(15.0)
+        expect(@interface.instance_variable_get(:@connect_timeout)).to eq(5.0)
+        expect(@interface.instance_variable_get(:@include_request_in_response)).to be true
+      end
+    end
+
+    describe "#connection_string" do
+      it "returns the correct URL" do
+        expect(@interface.connection_string).to eq('https://example.com:8080')
+      end
```

Comment:
Combine with existing ... the existing tests are more thorough

Reply:
Yes, done now.